magefiles/lint.go

```diff
@@ -73,7 +73,7 @@ func (Lint) Gofumpt() error {
 // Golangcilint Run golangci-lint
 func (Lint) Golangcilint() error {
 	fmt.Println("running golangci-lint")
-	return RunSh("go", WithV())("run", "github.com/golangci/golangci-lint/cmd/golangci-lint", "run", "--fix")
+	return RunSh("go", WithV())("run", "github.com/golangci/golangci-lint/v2/cmd/golangci-lint@v2.0.2", "run", "--fix")
```

Comment:
We are pinning the tool versions via `tools.go` and `go.mod`, so setting the version here is not appropriate, and we are also in turn not defining the v2 as a dependency in `tools.go`